moderngl/src/Context.cpp

```diff
@@ -269,15 +269,35 @@ PyObject * MGLContext_copy_framebuffer(MGLContext * self, PyObject * args) {
 			height = src->height < dst_framebuffer->height ? src->height : dst_framebuffer->height;
 		}
 
+		if (dst_framebuffer->draw_buffers_len != src->draw_buffers_len)
+		{
+			MGLError_Set("Destination and source framebuffers have different number of color attachments!");
+			return 0;
+		}
+
+
+		int prev_read_buffer = -1;
+		int prev_draw_buffer = -1;
+		int color_attachment_len = dst_framebuffer->draw_buffers_len;
+		gl.GetIntegerv(GL_READ_BUFFER, &prev_read_buffer);
+		gl.GetIntegerv(GL_DRAW_BUFFER, &prev_draw_buffer);
 		gl.BindFramebuffer(GL_READ_FRAMEBUFFER, src->framebuffer_obj);
 		gl.BindFramebuffer(GL_DRAW_FRAMEBUFFER, dst_framebuffer->framebuffer_obj);
-		gl.BlitFramebuffer(
-			0, 0, width, height,
-			0, 0, width, height,
-			GL_COLOR_BUFFER_BIT | GL_DEPTH_BUFFER_BIT,
-			GL_NEAREST
-		);
+
+		for (int i = 0; i < color_attachment_len; ++i)
+		{
+			gl.ReadBuffer(src->draw_buffers[i]);
+			gl.DrawBuffer(dst_framebuffer->draw_buffers[i]);
+			gl.BlitFramebuffer(
+				0, 0, width, height,
+				0, 0, width, height,
+				GL_COLOR_BUFFER_BIT | GL_DEPTH_BUFFER_BIT,
+				GL_NEAREST
+			);
+		}
 		gl.BindFramebuffer(GL_FRAMEBUFFER, self->bound_framebuffer->framebuffer_obj);
+		gl.ReadBuffer(prev_read_buffer);
+		gl.DrawBuffer(prev_draw_buffer);
```

Comment:
It has to be checked if this messes up theself->bound_framebuffer's `glDrawBuffers` call (please note the s at the end)
I assume gl.DrawBuffer(x) is equivalent of `glDrawBuffers(1, &{x})` so setting all the remaining attachments to GL_NONE.

Maybe we want https://github.com/moderngl/moderngl/blob/master/moderngl/src/Framebuffer.cpp#L405 instead

Reply:
That is an excellent point! 

That being said, OpenGL docs are not completely clear on the differences between glDrawBuffer and glDrawBuffers, esp. in the context of glBlitFramebuffer operation. I thought I could replace my calls to glDrawBuffers, but that results in an odd behaviour where all images show blue triangle. So it seems (empirically) that separate glDrawBuffers are needed for proper copying (?). 

I defensively added call to glDrawBuffers to restore the state stored in the `bound_framebuffer`. Let me know what you think.